src/services/review/RepoCheckerService.ts

```diff
@@ -493,52 +534,83 @@ export default class RepoCheckerService {
     return false
   }
 
-  reporter(repo: string, errors: RepoError[]): ResultAsync<RepoError[], never> {
-    const folderPath = path.join(SITE_CHECKER_REPO_PATH, repo)
-    if (!fs.existsSync(folderPath)) {
-      fs.mkdirSync(folderPath, { recursive: true })
-    }
-    const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
-    const stringifiedErrors = this.convertToCSV(errors)
-    if (!fs.existsSync(filePath)) {
-      fs.appendFileSync(filePath, stringifiedErrors)
-    } else {
-      fs.writeFileSync(filePath, stringifiedErrors)
-    }
+  reporter(
+    repo: string,
+    errors: RepoError[]
+  ): ResultAsync<RepoError[], SiteCheckerError> {
+    return this.createLogsFolder(repo).andThen(() => {
+      const filePath = path.join(SITE_CHECKER_REPO_PATH, repo, REPO_LOG)
+      const stringifiedErrors = this.convertToCSV(errors)
+      return ResultAsync.fromPromise<fs.Stats, false>(
+        fs.promises.stat(filePath),
+        () => false
```

Comment:
if it's the error branch, we discarded the error info to replace it w/ `false`, which is less descriptive than the error itself. perhaps we should let the caller decide this?

Reply:
hey @seaerchin, this one i might want to push back. 
this function is responsible for persisting the report to disk, the caller should not be concerned on whether there already existed a pervious run of the report already existing in the disk. This is just a I/O call to see if the file exists. It just so happens that the api call is designed such that this will throw an error if such file does not exist, which just means that we need to create the file in the `orElse` branch.

Reply:
thanks for this comment, after reading the documentation could just call write file directly for this, they recommend this approach instead